src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs

```diff
@@ -100,7 +100,11 @@ public Task<Stream> RequestServiceAsync(string serviceName)
             // this is what consumer actually use to communicate information
             var serviceStream = await _inprocServices.RequestServiceAsync(serviceName).ConfigureAwait(false);
 
-            return new JsonRpcConnection(Workspace, _inprocServices.Logger, callbackTarget, serviceStream, _remotableDataRpc.TryAddReference() ?? throw new ObjectDisposedException(GetType().FullName));
+            var remotableDataRpc = _remotableDataRpc.TryAddReference();
+            if (remotableDataRpc.IsDefault)
+                throw new ObjectDisposedException(GetType().FullName);
+
+            return new JsonRpcConnection(Workspace, _inprocServices.Logger, callbackTarget, serviceStream, remotableDataRpc.Move());
```

Comment:
📝 This code is more complex since `??` doesn't work with value types. The likely solution would be an API approach of defining an `AddReference()` method that throws `ObjectDisposedException` on failure.